examples/gno.land/p/demo/uuid/uuid.gno

```diff
@@ -0,0 +1,132 @@
+// Package uuid provides utility functions for generating unique identifiers
+// (UUIDs) based on a custom implementation inspired by the Snowflake algorithm.
+// The UUIDs generated by this package ensure uniqueness and are suitable for
+// distributed systems. The package offers a mechanism to convert these IDs
+// to standard UUID format strings, making them compatible with systems
+// requiring UUIDs.
+// add import "gno.land/p/demo/entropy" to use the entropy package. For of the pseudo-random number generator.
+
+package uuid
+
+import (
+	"encoding/binary"
+	"std"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/entropy"
+	"gno.land/p/demo/ufmt"
+)
+
+const (
+	TimestampLength uint8  = 41
+	MachineIDLength uint64 = 10
+	SequenceLength  uint64 = 12
+	MaxSequence     uint16 = 1<<SequenceLength - 1
+	MaxTimestamp    uint64 = 1<<TimestampLength - 1
+	MaxMachineID    uint64 = 1<<MachineIDLength - 1
+
+	machineIDMoveLength = SequenceLength
+	timestampMoveLength = MachineIDLength + SequenceLength
+)
+
+// UUID struct for generating unique identifiers.
+type UUID struct {
+	MachineID     uint64
+	sequence      uint16
+	StartTime     int64
+	LastTimestamp int64
+	entropy       *entropy.Instance
+}
+
+func NewUUID() *UUID {
+	u := &UUID{
+		StartTime: time.Date(2008, 11, 10, 23, 0, 0, 0, time.UTC).Unix() * 1000,
+		// counter:   0,
+		entropy: entropy.New(),
+	}
+	u.SetMachineID()
+	return u
+}
+
+// ID generates
+func (u *UUID) ID() uint64 {
+	current := u.incrementEntropy()
+	elapsedTime := current - u.StartTime
+	u.LastTimestamp = current
+
+	if uint64(elapsedTime) < MaxTimestamp {
```

Comment:
When is this possible?

Reply:
my logic is wrong, it should be `>`

But changing to `>` I ran into problems with the values overflowing too quickly because I was returning entropy in `uint64` whereas originally it was returning `uint32`. 

So for consistency of units between calculations I changed to `uint32`

What do you think?